python/lib/repository_standards.py

```diff
@@ -0,0 +1,201 @@
+    from the GitHub API and returns a report of the repository. This report is used to determine if the
+    repository is compliant with the standards set by Ops Engineering. It will be used to formulate a table
+    of repositories that are compliant and non-compliant.
+    """
+
+    def __init__(self, raw_github_data):
+        """
+        Initialise the class with the raw GitHub data.
+        """
+        self.repo_data = raw_github_data
+        self.report_output = self.__generate_report()
+        self.repository_type = "private" if self.__is_private() else "public"
+
+    def __generate_report(self) -> dict:
+        """
+        Generate the report for the repository, used to determine if the repository is compliant or not.
+        """
+        return {
+            "name": self.__repo_name(),
+            "default_branch": self.__default_branch(),
+            "url": self.__url(),
+            "status": self.is_compliant(),
+            "report": self.compliance_report(),
+            "last_push": self.__last_push(),
+            "is_private": self.__is_private()
+        }
+
+    def __repo_name(self) -> str:
+        return self.repo_data["node"]["name"]
+
+    def __default_branch(self) -> str:
+        return self.repo_data["node"]["defaultBranchRef"]["name"]
+
+    def __url(self) -> str:
+        return self.repo_data["node"]["url"]
+
+    def is_compliant(self) -> bool:
+        """
+        Returns True if the repository is compliant, False otherwise.
+        """
+        for key, value in self.compliance_report().items():
+            if value is False:
+                print(f"Non-compliant: {key} is {value}")
+                return False
+
+        return True
+
+    def __last_push(self) -> str:
+        return self.repo_data["node"]["pushedAt"]
+
+    def __is_private(self) -> bool:
+        return self.repo_data["node"]["isPrivate"]
+
+    def compliance_report(self) -> dict:
+        """
+        Returns a dictionary of the compliance report for the repository. The report is an opinionated
+        view of what is compliant and what is not. This is based on the standards set by Ops Engineering.
+        """
+        return {
+            "default_branch_main": self.__default_branch_main(),
+            "has_default_branch_protection": self.__has_default_branch_protection_enabled(),
+            "requires_approving_reviews": self.__has_requires_approving_reviews_enabled(),
+            "administrators_require_review": self.__has_admin_requires_reviews_enabled(),
+            "issues_section_enabled": self.__has_issues_enabled(),
+            "has_require_approvals_enabled": self.__has_required_approval_review_count_enabled(),
+            "has_license": self.__has_license(),
+            "has_description": self.__has_description()
+        }
+
+    def __default_branch_main(self) -> bool:
+        if self.repo_data["node"]["defaultBranchRef"]["name"] == "main":
```

Comment:
Love the small scope functions 💖

I might be wrong, would this also work?

```python
def __default_branch_main(self) -> bool:
        return self.repo_data["node"]["defaultBranchRef"]["name"] == "main"
```

Reply:
Done.